### DISCUSSION TITLE
Asynchronous issue

### DISCUSSION THREAD

**Discussion by zhangtingwz** [elsa-workflows/elsa-core#2692] [Q&A]
hi,nice to meet u ,friend.
i meet a question that i use 
 `        public async Task<IEnumerable<CollectedWorkflow>> DispatchWorkflowsAsync(
            CancellationToken cancellationToken = default,
            string WorkflowInstanceId = null,
            string CorrelationId = null,
            WorkflowInput input = null,
            int type = 0)
        {
            var context = new WorkflowsQuery(
                            type == 0 ? nameof(SchoolApprovalAction):nameof( FunctionalDepartmentAction), 
                            new SchoolApprovalBookmark(),
                            CorrelationId, 
                            WorkflowInstanceId);
            return await _workflowLaunchpad.CollectAndDispatchWorkflowsAsync(context, input, cancellationToken);
        }`
After the method is executed, the method in my action is executed. Because the database data is modified in the action, the display on the page is different from that in the database.
please help me

**Comment by zhangtingwz** [comment]
then i change CollectAndDispatchWorkflowsAsync => CollectAndDispatchWorkflowsAsync it is wait action is executed

**Comment by sfmskywalker** [comment]
Hi @zhangtingwz , nice to meet you!

That makes sense. The `DispatchWorkflowsAsync` method will not actually execute the workflow. Instead, it simply sends a message to a queue and returns immediately. A separate background worker will then execute the workflows.

However, you can totally execute the workflow synchronously by using the `ExecuteWorkflowsAsync` method instead - that one will first execute the workflows before returning to the caller.
